window/src/os/macos/app.rs

```diff
@@ -68,27 +69,27 @@ extern "C" fn application_will_finish_launching(
     log::debug!("application_will_finish_launching");
 }
 
-extern "C" fn application_did_finish_launching(this: &mut Object, _sel: Sel, _notif: *mut Object) {
+extern "C" fn application_did_finish_launching(_self: &mut Object, _sel: Sel, _notif: *mut Object) {
     log::debug!("application_did_finish_launching");
-    unsafe {
-        (*this).set_ivar("launched", YES);
```

Comment:
FWIW, I suspect that removing this launched flag is probably the thing that's causing your multiple window issue